image_tools/src/cam2image.cpp

```diff
@@ -71,92 +67,91 @@ void convert_frame_to_message(
-      &height, &burger_mode, &topic))
-  {
-    return 0;
+  if (setup(argc, argv)) {
+    execute();
+  } else {
+    rclcpp::shutdown();
+    return;
   }
+}
 
-  // Initialize a ROS 2 node to publish images read from the OpenCV interface to the camera.
-  auto node = rclcpp::Node::make_shared("cam2image");
-  rclcpp::Logger node_logger = node->get_logger();
+/// Read in and parse command line arguments.
+/**
+ * \param[in] argc
+ * \param[in] argv
+ * \return A bool whether command line options were valid or not
+ */
+bool Cam2Image::setup(int argc, char ** argv)
+{
+  return parse_command_options(
+    argc, argv, &depth_, &reliability_policy_, &history_policy_, &show_camera_, &freq_, &width_,
+    &height_, &burger_mode_, &topic_);
+}
 
-  // Set the parameters of the quality of service profile. Initialize as the default profile
-  // and set the QoS parameters specified on the command line.
+/// Execute main functions of component
+/**
+ * publish camera feed or burger image to "image" topic
+ */
+void
+Cam2Image::execute()
+{
+  rclcpp::Logger node_logger = this->get_logger();
   auto qos = rclcpp::QoS(
     rclcpp::QoSInitialization(
-      // The history policy determines how messages are saved until taken by the reader.
-      // KEEP_ALL saves all messages until they are taken, up to a system resource limit.
-      // KEEP_LAST enforces a limit on the number of messages that are saved.
-      // The limit is specified by the history "depth" parameter.
-      history_policy,
-      // Depth represents how many messages to save in history when the history policy is KEEP_LAST.
-      depth));
-
+      // The history policy determines how messages are saved until taken by
+      // the reader.
+      // KEEP_ALL saves all messages until they are taken.
+      // KEEP_LAST enforces a limit on the number of messages that are saved,
+      // specified by the "depth" parameter.
+      history_policy_,
+      // Depth represents how many messages to store in history when the
+      // history policy is KEEP_LAST.
+      depth_
+  ));
   // The reliability policy can be reliable, meaning that the underlying transport layer will try
   // ensure that every message gets received in order, or best effort, meaning that the transport
   // makes no guarantees about the order or reliability of delivery.
-  qos.reliability(reliability_policy);
-
-  RCLCPP_INFO(node_logger, "Publishing data on topic '%s'", topic.c_str());
-  // Create the image publisher with our custom QoS profile.
-  auto pub = node->create_publisher<sensor_msgs::msg::Image>(topic, qos);
+  qos.reliability(reliability_policy_);
+  RCLCPP_INFO(node_logger, "Publishing data on topic '%s'", topic_.c_str());
+  pub_ = create_publisher<sensor_msgs::msg::Image>(topic_, qos);
 
   // is_flipped will cause the incoming camera image message to flip about the y-axis.
   bool is_flipped = false;
-
-  // Subscribe to a message that will toggle flipping or not flipping, and manage the state in a
```

Comment:
nit: I'd leave this comment in since it is demo code we expect new users to look at.